data-prepper-plugins/peer-forwarder/src/main/java/com/amazon/dataprepper/plugins/processor/peerforwarder/HashRing.java

```diff
@@ -0,0 +1,64 @@
+package com.amazon.dataprepper.plugins.processor.peerforwarder;
+
+import javax.annotation.Nonnull;
+import javax.annotation.concurrent.NotThreadSafe;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.zip.CRC32;
+import java.util.zip.Checksum;
+
+@NotThreadSafe
+public class HashRing {
+    private final List<String> serverIps = new ArrayList<>();
+    private final int numVirtualNodes;
+    private final TreeMap<Long, String> virtualNodes = new TreeMap<>();
+
+    public HashRing(@Nonnull final List<String> serverIps, final int numVirtualNodes) {
+        this.numVirtualNodes = numVirtualNodes;
+        final List<String> sortedServerIps = new ArrayList<>(serverIps);
+        Collections.sort(sortedServerIps);
```

Comment:
Not necessarily. My initial intention is to make sure it produces the same virtual nodes for the same set of IPs if there is a hash collision. But it might not be a necessary concern.